src/zero_copy/combinator.rs

```diff
@@ -1178,18 +1219,19 @@ impl<A, B, C, const N: usize> SeparatedByExactly<A, B, C, N> {
     }
 }
 
-impl<'a, I, E, S, A, B, C, const N: usize> Parser<'a, I, E, S> for SeparatedByExactly<A, B, C, N>
+// FIXME: why parser output is not C ?
+impl<'a, I, E, S, A, B, OA, OB, C, const N: usize> Parser<'a, I, [OA; N], E, S>
+    for SeparatedByExactly<A, B, OB, C, N>
 where
     I: Input + ?Sized,
     E: Error<I>,
     S: 'a,
-    A: Parser<'a, I, E, S>,
-    B: Parser<'a, I, E, S>,
-    C: ContainerExactly<A::Output, N>,
+    A: Parser<'a, I, OA, E, S>,
+    B: Parser<'a, I, OB, E, S>,
+    C: ContainerExactly<OA, N>,
 {
-    type Output = [A::Output; N];
-
-    fn go<M: Mode>(&self, inp: &mut InputRef<'a, '_, I, E, S>) -> PResult<M, Self::Output, E> {
+    // FIXME: why parse result output is not C ?
```

Comment:
See above

Reply:
this an unrelated FIXME as far as I can tell

Reply:
Oh, good point. Yes, the output type should be `C` I believe!

Reply:
Changing the output type to C would require modification of unsafe code, which I have never done. I can just drop the the general container type `C` for the array type or someone else has to fix this.

Reply:
I can fix it once merged :)